### PR TITLE
add top bar for TV grid in Template's form

### PR DESCRIPTION
**Pull request by goldsky**
### What does it do ?

Adding filters (by category or searching) for TV grid in Template edit mode

tvtemplatetopbar
### Why is it needed ?

Number of Template Variables can raise to hundreds on a big site, especially for multilingual sites.
Having filter helps out the selection.

### CODE REVIEW

**Comment by theboxer** [comment]
There's contribution guide (https://github.com/modxcms/revolution/blob/2.x/CONTRIBUTING.md), so please update the PR to respect it.

Also the posted screenshot has nothing to do with this PR and is missleading. Changes are about TV panel, not Template panel.

When testing the top bar, filtering and search stops working when you press clear filter button (lead to processor not found error)

File: manager/assets/modext/widgets/element/modx.grid.tv.template.js [hunk 1]
```diff
@@ -35,12 +35,74 @@ MODx.grid.TemplateVarTemplate = function(config) {
...
+        }]
     });
     MODx.grid.TemplateVarTemplate.superclass.constructor.call(this,config);
 };
-Ext.extend(MODx.grid.TemplateVarTemplate,MODx.grid.Grid);
+Ext.extend(MODx.grid.TemplateVarTemplate,MODx.grid.Grid,{
+    filterByCategory: function(cb,rec,ri) {
+        this.getStore().baseParams['category'] = cb.getValue();
+        this.getBottomToolbar().changePage(1);
+        this.refresh();
```
Comment (theboxer) [inline manager/assets/modext/widgets/element/modx.grid.tv.template.js]: There's no need for calling `refresh()`, `changePage(1)` calls it, so you're calling it twice which is unnecessary and produces canceled request.

File: manager/assets/modext/widgets/element/modx.grid.tv.template.js [hunk 2]
```diff
@@ -35,12 +35,74 @@ MODx.grid.TemplateVarTemplate = function(config) {
...
+    }
+    ,search: function(tf,newValue,oldValue) {
+        var nv = newValue || tf;
+        this.getStore().baseParams.query = Ext.isEmpty(nv) || Ext.isObject(nv) ? '' : nv;
+        Ext.getCmp('modx-tvtemp-filter-category').setValue('');
+        this.getBottomToolbar().changePage(1);
+        this.refresh();
```
Comment (theboxer) [inline manager/assets/modext/widgets/element/modx.grid.tv.template.js]: There's no need for calling `refresh()`, `changePage(1)` calls it, so you're calling it twice which is unnecessary and produces canceled request.

File: manager/assets/modext/widgets/element/modx.grid.tv.template.js [hunk 3]
```diff
@@ -35,12 +35,74 @@ MODx.grid.TemplateVarTemplate = function(config) {
...
+        return true;
+    }
+    ,clearFilter: function() {
+    	this.getStore().baseParams = {
+            action: 'getList'
+    	};
+        Ext.getCmp('modx-tvtemp-filter-category').reset();
+        Ext.getCmp('modx-tvtemp-search').setValue('');
+    	this.getBottomToolbar().changePage(1);
+        this.refresh();
```
Comment (theboxer) [inline manager/assets/modext/widgets/element/modx.grid.tv.template.js]: There's no need for calling `refresh()`, `changePage(1)` calls it, so you're calling it twice which is unnecessary and produces canceled request.

File: manager/assets/modext/widgets/element/modx.grid.tv.template.js [hunk 4]
```diff
@@ -35,12 +35,74 @@ MODx.grid.TemplateVarTemplate = function(config) {
...
+        var nv = newValue || tf;
+        this.getStore().baseParams.query = Ext.isEmpty(nv) || Ext.isObject(nv) ? '' : nv;
+        Ext.getCmp('modx-tvtemp-filter-category').setValue('');
+        this.getBottomToolbar().changePage(1);
+        this.refresh();
+        return true;
+    }
+    ,clearFilter: function() {
+    	this.getStore().baseParams = {
+            action: 'getList'
```
Comment (theboxer) [inline manager/assets/modext/widgets/element/modx.grid.tv.template.js]: Produces processor not found error, should be `element/tv/template/getList`

**Comment by theboxer** [comment]
Please test the PR before sending it...

**Comment by goldsky** [comment]
I did, using 2.5.0-dev.
Ok, Ill update the info

**Comment by goldsky** [comment]
description has been updated

**Pull request merged**